### PR TITLE
6 booking request

### LINKED ISSUE #6: booking request
```
As a user
So I can stay in a space
I want to make a booking request for a space
```

### PR DESCRIPTION
**Pull request by tobenna**
Closes #6

### CODE REVIEW

File: app/controllers/requests.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class MakersBnb < Sinatra::Base
+  get '/requests/:space_id' do
```
Comment (kevinpmcc) [inline app/controllers/requests.rb]: move this to spaces controller please. Have a look at mockups 
/spaces/:space_id

File: app/controllers/requests.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    @page_title = 'Book a Space'
+    erb(:'requests/new')
+  end
+
+  post '/requests' do
+    request_details = { user: current_user, date: params[:date] }
+    req = Request.new(request_details)
+    req.space = Space.first(id: params[:space_id])
+    req.save
+    redirect('/requqest/:space_id')
```
Comment (adilw3nomad) [inline app/controllers/requests.rb]: request not requqest

File: app/controllers/spaces.rb [hunk 3]
```diff
@@ -7,9 +7,15 @@ class MakersBnb < Sinatra::Base
 
   post '/spaces' do
     @space = Space.new(name: params[:name],
-                        price_per_night: params[:"price-per-night"],
```
Comment (adilw3nomad) [inline app/controllers/spaces.rb]: too many indentions, please fix

File: spec/features/booking_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  end
+
+  scenario 'making a request' do
+    sign_up
+    visit '/requests/1'
+    fill_in(:date, with: '2017/04/21')
+    request = 'Request to Book'
+    expect { click_button(request) }.to change { Request.all.size }.by 1
+  end
+end
```
Comment (adilw3nomad) [inline spec/features/booking_request_spec.rb]: maybe we could add tests for validation before we close this issue

Comment (adilw3nomad) [inline spec/features/booking_request_spec.rb]: tests for input validation



# TOBENNA

File: spec/features/new_space_spec.rb [hunk 5]
```diff
@@ -2,12 +2,12 @@
   scenario 'space is made' do
     visit '/spaces/new'
     fill_in('name', with: 'mySpace')
-    fill_in('description', with: 'Jelly macaroon toffee.')
```
Comment (adilw3nomad) [inline spec/features/new_space_spec.rb]: i liked jelly macaroon toffee better 

# TOBENNA

**Comment by kevinpmcc** [comment]
create issue to fix date validation after mvp

**Comment by kevinpmcc** [comment]
:shipit:

**Pull request merged**